### PR TITLE
Use 'uploadDataDuringCreation' option when uploading via tus

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Uploading via tus now uses the `uploadDataDuringCreation` option which saves up one request. Also, we've fixed a serialization error during uploads, saving up another request.

## Related Issue
- https://github.com/owncloud/web/issues/7066

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

### CODE REVIEW

File: packages/web-runtime/src/composables/upload/useUpload.ts [hunk 1]
```diff
@@ -9,7 +9,6 @@ import { computed, Ref, unref, watch } from '@vue/composition-api'
 import { useActiveLocation } from 'files/src/composables'
 import { isLocationPublicActive } from 'files/src/router'
 import { UppyService } from '../../services/uppyService'
-import { Location } from 'vue-router/types/router'
 import * as uuid from 'uuid'
 
 export interface UppyResource {
```
Comment (kulmann) [inline packages/web-runtime/src/composables/upload/useUpload.ts:14]: Could you please add a comment that this interface MUST never contain complex types / MUST always consist of primitive types only? With the serialization reasoning. I guess that's a fair warning that should stay in the code to not run into the same trap again in the future.

File: packages/web-runtime/src/composables/upload/useUpload.ts [hunk 2]
```diff
@@ -22,11 +21,15 @@ export interface UppyResource {
...
-    route: Location
     tusEndpoint: string
     webDavBasePath: string
     uploadId: string
     topLevelFolderId?: string
+    routeName?: string
+    routeItem?: string
+    routeShareName?: string
+    routeStorage?: string
+    routeStorageId?: string
```
Comment (kulmann) [inline packages/web-runtime/src/composables/upload/useUpload.ts:32]: For the share jail / `SharedResource.vue` we also need the `shareId` as additional param. That doesn't come from `route.params` but `route.query` though.

File: changelog/unreleased/enhancement-upload-data-during-creation [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Enhancement: Upload data during creation
```
Comment (kulmann) [inline changelog/unreleased/enhancement-upload-data-during-creation:1]: I think the bugfix is worth a separate changelog item. 🚀

**Review by kulmann — CHANGES_REQUESTED** [review]

File: packages/web-runtime/src/services/uppyService.ts [hunk 4]
```diff
@@ -51,7 +51,8 @@ export class UppyService {
       chunkSize: chunkSize,
       removeFingerprintOnSuccess: true,
       overridePatchMethod: !!tusHttpMethodOverride,
-      retryDelays: [0, 500, 1000]
+      retryDelays: [0, 500, 1000],
+      uploadDataDuringCreation: true
```
Comment (kulmann) [inline packages/web-runtime/src/services/uppyService.ts:55]: can we use capabilities here? the tus plugin is announced in the capabilities AFAIR.

**Review by kulmann — APPROVED** [review]
Nice 💪

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/26156/12/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/26156/tracing/unstructured-collection-of-testable-space-interactions-alice-2022-6-13-08-46-58.zip 


</p></details>

**Pull request merged**